src/formatDate.js

```diff
@@ -8,7 +8,70 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  if (date !== undefined) {
+    const valuesOfDate = date.split(fromFormat[3]);
+
+    let DD = 0;
+    let MM = 0;
+    let year = 0;
+    let result = '';
+
+    for (let i = 0; i <= fromFormat.length - 2; i++) {
+      switch (fromFormat[i]) {
+        case 'YYYY':
+          year = valuesOfDate[i];
+          break;
+        case 'YY':
+          year = valuesOfDate[i];
+          break;
+        case 'MM':
+          MM = valuesOfDate[i];
+          break;
+        case 'DD':
+          DD = valuesOfDate[i];
+          break;
+      }
+    }
+
+    for (let i = 0; i <= toFormat.length - 2; i++) {
+      switch (toFormat[i]) {
+        case 'YYYY':
+          if (year.length === 4) {
+            result += year;
+          } else {
+            if (parseInt(year, 10) > 24) {
+              year = `19${year}`;
+              result += year;
+            } else {
+              year = `20${year}`;
+              result += year;
+            }
```

Comment:
The condition `parseInt(year, 10) > 24` should be `parseInt(year, 10) >= 30` according to the task description. This is because years less than `30` should be prefixed with `20`, and `30` or greater should be prefixed with `19`.